File: sdk/go/common/util/cmdutil/testdata/term_wait.js
function handleSignal() {
	console.log('exiting cleanly');
	process.exit(0);
}

process.on('SIGINT', handleSignal);
console.log('ready');

setTimeout(function() {
	console.error('error: did not receive signal');
	process.exit(1);
}, 3000);

File: sdk/go/common/util/cmdutil/testdata/term_wait.go
package main

import (
	"fmt"
	"os"
	"os/signal"
	"time"
)

func main() {
	sigch := make(chan os.Signal, 1)
	signal.Notify(sigch, os.Interrupt)
	// os.Interrupt handles SIGINT and CTRL_BREAK_EVENT.

	fmt.Println("ready")
	select {
	case <-sigch:
		fmt.Println("exiting cleanly")
		os.Exit(0)

	case <-time.After(3 * time.Second):
		fmt.Fprintln(os.Stderr, "error: did not receive signal")
		os.Exit(1)
	}
}
